### PR TITLE
docs(performance): Add performance config documentation

### LINKED ISSUE #435: Document new performance budgets feature
Now that most of the details are stamped out for https://github.com/webpack/webpack/pull/3350#pullrequestreview-9743611, we need to ensure that we are documenting the new configuration features, as well as specify what they represent. 

Here is the schema for the new information: 

```
    "performance": {
      "description": "Configuration for web performance recommendations.",
      "additionalProperties": false,
      "properties": {
        "hints": {
          "description": "Turn hints on or off",
          "type": "boolean"
        },
        "maxInitialChunkSize": {
          "description": "Total size of all initial chunks (in bytes)",
          "type": "number"
        },
        "maxAssetSize": {
          "description": "Filesize limit (in bytes) when exceeded, that webpack will provide performance hints",
          "type": "number"
        },
        "errorOnHint": {
          "description": "Throw errors instead of warnings when performance hints are found",
          "type": "boolean"
        }
      },
      "type": "object"
    }
```

Performance property: 
Default recommendations about web practice and performance. 

Reference https://github.com/webpack/webpack/issues/3216 for more details.

### PR DESCRIPTION
**Pull request by TheLarkInn**
Fixes #435

### CODE REVIEW

File: content/configuration/performance.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+---
+title: Performance
+sort: 14
+contributors:
+  - thelarkinn
+---
+
+The options allows you to control how webpack notifies you of assets and entrypoints that exceed a specific file limit.
```
Comment (SpaceK33z) [inline content/configuration/performance.md]: s/The/These/

File: content/configuration/performance.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+No hint warnings or errors are shown.
+
+```js
+performance: {
+  hints: "warning"
+}
+```
+
+A warning will be displayed notifying you of a large asset. We recommend something like this for development environments.
```
Comment (bebraw) [inline content/configuration/performance.md]: Let's say you are inlining assets during development, missing minification etc., wouldn't this yield warnings even if they aren't relevant? Isn't the best value in production build?

Comment (TheLarkInn) [inline content/configuration/performance.md]: Yes, but also gives perspective to users as they are developing. 

If a bundle grows from 200->350kb after a certain feature, I'd want to know specifically about it while developing so I can trace my steps back a few changes and investigate more into what is causing the bloat.

Comment (bebraw) [inline content/configuration/performance.md]: Alright. I guess it takes more experience with the feature for me to tell any better. Hitting the limits is very easy by inlining just a few images.

Comment (TheLarkInn) [inline content/configuration/performance.md]: Yes true. Maybe that's where they could filter assets, etc.

I mean at the most these are recommendations but I 100% agree where you are coming from.

Comment (bebraw) [inline content/configuration/performance.md]: Would it make sense to have something like `--json` kind of flag for the perf data? Can you get it in JSON to track?

Comment (TheLarkInn) [inline content/configuration/performance.md]: Aww that would actually be a really cool feature but the log might be built. I'm going to add that to the original issue.

Comment (TheLarkInn) [inline content/configuration/performance.md]: Might be big*

**Comment by TheLarkInn** [comment]
Any other looks? Want to get this in since Perf is in now.

**Comment by TheLarkInn** [comment]
Any other objections? Going to merge this.

**Comment by bebraw** [comment]
Yeah, merge. More tweaks can be done in separate PRs.

**Pull request merged**